File: fixtures/scale/scala/medium/src/calc/funcs/Conversion.scala
package calc.funcs

import calc.eval.FunctionRegistry

object Conversion {
  def register(registry: FunctionRegistry): Unit = {
    registry.addUnary("radToDeg", math.toDegrees)
    registry.addUnary("degToRad", math.toRadians)
    registry.addUnary("percent", _ / 100.0)
    registry.addAggregate("percentOf", args => args(0) / 100.0 * args(1))
  }
}

File: fixtures/scale/kotlin/medium/src/calc/funcs/Conversion.kt
package calc.funcs

import calc.eval.FunctionRegistry

fun registerConversion(registry: FunctionRegistry) {
    registry.addUnary("radToDeg") { Math.toDegrees(it) }
    registry.addUnary("degToRad") { Math.toRadians(it) }
    registry.addUnary("percent") { it / 100.0 }
    registry.addAggregate("percentOf") { it[0] / 100.0 * it[1] }
}
